db/channel_cache.go

```diff
@@ -167,9 +167,10 @@ func (c *channelCache) _getCachedChanges(options ChangesOptions) (validFrom uint
 	}
 
 	n := len(log) - start
-	if options.Limit > 0 && n > options.Limit {
+	if options.Limit > 0 && n > options.Limit && !options.ActiveOnly {
```

Comment:
I don't know why we need to do the ActiveOnly check here.  Can you add a comment to clarify?